### PR TITLE
fix(cli): inconsistent behavior across CLI flags, environment variables, and config files

### PR DESCRIPTION
**Pull request by knqyf263**
## Description
In a recent update to Trivy, we introduced [validation](https://github.com/aquasecurity/trivy/pull/4800) and [alias](https://github.com/aquasecurity/trivy/pull/5558) functionalities for CLI flags. However, we realized that these enhancements had not been uniformly applied to other configuration methods, specifically environment variables and the configuration file (trivy.yaml). This PR addresses this oversight by extending the validation and alias features across all configuration settings, ensuring consistency and improved usability.

To facilitate this integration, we've added a `Parse()` method. This method is designed to be called before accessing values through the `Value()` method, mirroring the interface pattern found in the Go's standard `flag` package ([flag package documentation](https://pkg.go.dev/flag#Parse)). This addition ensures that value retrieval does not result in errors and that the parsing process is streamlined across different settings.

Additionally, we've refactored the handling of default flag values. Previously, the logic was branched based on the type of the `Default` field of each flag. We've now shifted this approach to leverage Go's generics, allowing for a more explicit and type-safe representation of default values.

## Related issues
- Close https://github.com/aquasecurity/trivy/issues/5803

## Checklist
- [x] I've read the [guidelines for contributing](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/) to this repository.
- [x] I've followed the [conventions](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/#title) in the PR title.
- [x] I've added tests that prove my fix is effective or that my feature works.
- [ ] I've updated the [documentation](https://github.com/aquasecurity/trivy/blob/main/docs) with the relevant information (if needed).
- [ ] I've added usage information (if the PR introduces new options)
- [ ] I've included a "before" and "after" example to the description (if the PR is a user interface change).

### CODE REVIEW

**Comment by knqyf263** [comment]
@naortalmor1 @tonaim I believe it requires small changes on your end, but please let me know if it has a big impact. We'll consider another approach in that case.

**Review by DmitriyLewen — COMMENTED** [review]
LGTM.
Left nit comment.

File: integration/config_test.go [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+			args: args{
+				input: "testdata/fixtures/repo/gomod",
+				flags: map[string]string{
+					"scanners":   "vuln",
+					"skip-files": "path/to/dummy,testdata/fixtures/repo/gomod/submod2/go.mod",
+				},
+				envs: map[string]string{
+					"TRIVY_SCANNERS":   "vuln",
+					"TRIVY_SKIP_FILES": "path/to/dummy,testdata/fixtures/repo/gomod/submod2/go.mod",
+				},
```
[the remark this answers is not in the document]
Comment (knqyf263) [reply integration/config_test.go:43]: It runs the test three times with CLI, envs, and the config file.
https://github.com/aquasecurity/trivy/blob/554d6fc001ef1ac7a1ac78283842c68340d77fa6/integration/config_test.go#L153-L228

So, flags and envs are not used at the same time.

[the remark this answers is not in the document]
Comment (DmitriyLewen) [reply integration/config_test.go:43]: I don't know how i missed this 😄 
Thanks for answer!

**Review by DmitriyLewen — APPROVED** [review]

**Pull request merged**